ml/jml/onevsall_generator.cc

```diff
@@ -0,0 +1,168 @@
+// This file is part of MLDB. Copyright 2017 mldb.ai inc. All rights reserved.
+
+/* onevsall_generator.cc                                          -*- C++ -*-
+   Mathieu Marquis Bolduc, 8 March 2017
+   Copyright (c) 2017 MLDB.ai  All rights reserved.
+   $Source$
+
+   Generator for onevsall classifiers
+*/
+
+#include "onevsall_generator.h"
+#include "mldb/ml/jml/registry.h"
+#include "mldb/jml/utils/smart_ptr_utils.h"
+#include "mldb/utils/log_fwd.h"
+#include "mldb/base/parallel.h"
+#include "mldb/ml/jml/info_override_feature_space.h"
+#include "mldb/ml/jml/multilabel_training_data.h"
+
+#include "training_index.h"
+
+using namespace std;
+
+
+namespace ML {
+
+/*****************************************************************************/
+/* ONEVSALL_GENERATOR                                                        */
+/*****************************************************************************/
+
+OneVsAll_Classifier_Generator::
+OneVsAll_Classifier_Generator()
+{
+    defaults();
+}
+
+OneVsAll_Classifier_Generator::
+OneVsAll_Classifier_Generator(std::shared_ptr<Classifier_Generator> learner)
+{
+    defaults();
+    weak_learner = learner;
+}
+
+OneVsAll_Classifier_Generator::~OneVsAll_Classifier_Generator()
+{
+}
+
+void
+OneVsAll_Classifier_Generator::
+configure(const Configuration & config, vector<string> & unparsedKeys)
+{
+    Classifier_Generator::configure(config, unparsedKeys);
+
+    config.find(verbosity, "verbosity");
+
+    weak_learner = get_trainer("weak_learner", config);
+}
+
+void
+OneVsAll_Classifier_Generator::
+defaults()
+{
+    Classifier_Generator::defaults();
+    weak_learner.reset();
+}
+
+Config_Options
+OneVsAll_Classifier_Generator::
+options() const
+{
+    Config_Options result = Classifier_Generator::options();
+    result       
+        .subconfig("weak_leaner", weak_learner,
```

Comment:
weak_learner has a very specific meaning for boosting and bagging, it's probably not the right name here.